### DISCUSSION TITLE
Allow enforcing clickhouse settings on explore queries

### DISCUSSION THREAD

**Discussion by tanner-bruce** [grafana/clickhouse-datasource#693] [Ideas]
Some things that could be helpful to force at the explore query level

https://clickhouse.com/docs/en/operations/settings/settings#force_index_by_date

https://clickhouse.com/docs/en/operations/settings/settings#force-primary-key

https://clickhouse.com/docs/en/operations/settings/settings#force_data_skipping_indices

**Comment by SpencerTorres** [comment]
This is an interesting idea, and not the first time I've seen people requesting the ability to control `SETTINGS` on the query builder. (Example https://github.com/grafana/clickhouse-datasource/issues/520#issuecomment-1723583880)

If you want, we can continue this as a feature request on the issues board. Feel free to submit an issue with an idea for how you would want this to be controlled via the UI or datasource config
